spec/support/mock_generator.rb

```diff
@@ -0,0 +1,12 @@
+# Used in generator_example_helpers_spec.rb
+# set_shell_prompt_responses requires a generator w/ a shell
+# as an argument. This class defines that generator w/ a shell.
+require_relative 'mock_shell'
+
+class Generator
+  attr_accessor :shell
+
+  def initialize
+    @shell = Shell.new
```

Comment:
So I tried this out, and it looks like Ruby actually has a problem with dynamically creating named classes in initializers like this. It throws me a `dynamic constant assignment` error message when I try this. Since the Shell technically has no real functionality, but more so the MockGenerator needs an object as it's `@shell` attribute, I simple changed the initialize method to look something like this:

```ruby
def initialize
    shell = Class.new
    @shell = shell.new
  end
```
That appears to be working